### PR TITLE
add envelope process result to return ccd id and type

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1191

### Change description ###

add envelope process result to return ccd id and type
next PR will process the result for ProcessedEnvelopeNotifier.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java [hunk 1]
```diff
@@ -42,33 +46,32 @@ public void handleEnvelope(Envelope envelope) {
...
                             existingCase.getId()
                         );
-                        createExceptionRecord(envelope);
                     }
-                } else {
-                    createExceptionRecord(envelope);
                 }
-                break;
+
+                return  new EnvelopeProcessResult(createExceptionRecord(envelope),  EXCEPTION_RECORD);
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java:59]: let's move that to two places above

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java:59]: also, 2 spaces after `return`

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.processedenvelopes;
+
+public enum ProcessedCcdType {
+    AUTO_ATTACHED_CASE,
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java:4]: *to* case

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.processedenvelopes;
+
+public enum ProcessedCcdType {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java:3]: maybe `EnvelopeCcdAction` or `EnvelopeCcdResult`?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.processedenvelopes;
+
+public class EnvelopeProcessResult {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java:3]: I think it should be 'processing'

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java:4]: this not for action, actually case or exception

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java (hunk 3 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java:3]: in processor we used  ProcessedCcdType, but  EnvelopeCcdAction sound good to me

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java (hunk 4 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java:3]: but it is end product, it should finished, completed successfully

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java:59]: > let's move that to two places above

one?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java:59]: line 51 and 54

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java:4]: I mean that we are not attaching a case. we are attaching something TO a case.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java:4]: but the type is case? affected type is case?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java]: I think this is clearer

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java]: > also, 2 spaces after `return`

removed

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java (hunk 3 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java]: changed to EnvelopeCcdAction

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java [hunk 5]
```diff
@@ -42,33 +46,31 @@ public void handleEnvelope(Envelope envelope) {
...
                     } else {
                         log.info(
                             "Creating exception record as supplementary evidence failed for envelope {} case {}",
                             envelope.id,
                             existingCase.getId()
                         );
-                        createExceptionRecord(envelope);
                     }
-                } else {
-                    createExceptionRecord(envelope);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java:54]: Why the removal of this line and also the line above?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java (hunk 4 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java:3]: Theoretically, processing is correct, but given we've got `ProcessResult` already for callbacks, makes sense to keep names consistent.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+    public final Long ccdId;
+    public final EnvelopeCcdAction processedCcdType;
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java:6]: `ccdActionType`? Not strong about it, just to apply or reject.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java:6]: ccdAction ?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java (hunk 4 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java:3]: I am making it Processing

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java]: I think Rafał treats this enum as describing CCD action type, while you're treating it as one describing the kind of CCD case. So maybe it should be one of the two or sth similar?
```
ATTACH_DOCUMENTS
CREATE_EXCEPTION_RECORD
```
or
```
SERVICE_CASE
EXCEPTION_RECORD
```
depending on how we look at it.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/ProcessedCcdType.java]: I think this field just defining  what ID is referencing. ID represents - a case or exception record.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java (hunk 4 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/processedenvelopes/EnvelopeProcessResult.java]: when it is Processing I expect the class to cover all results, exceptions everything and return to me as result.but we just return this class if it is scuccessfull.

**Pull request merged**